File: app/src/main/java/by/toggi/rxbsuir/RxBsuirApplication.java
package by.toggi.rxbsuir;

import android.app.Application;
import android.content.SharedPreferences;

import androidx.appcompat.app.AppCompatDelegate;

import javax.inject.Inject;

import dagger.hilt.android.HiltAndroidApp;

@HiltAndroidApp
public class RxBsuirApplication extends Application implements SharedPreferences.OnSharedPreferenceChangeListener {

    @Inject
    SharedPreferences mSharedPreferences;

    @Override
    public void onCreate() {
        super.onCreate();
        mSharedPreferences.registerOnSharedPreferenceChangeListener(this);
    }

    @Override
    public void onSharedPreferenceChanged(SharedPreferences sharedPreferences, String key) {
        if (getString(R.string.night_mode_key).equals(key)) {
            var nightModeString = sharedPreferences.getString(key, getString(R.string.night_mode_default_value));
            var nightMode = Integer.parseInt(nightModeString);
            AppCompatDelegate.setDefaultNightMode(nightMode);
        }
    }
}

File: app/src/main/kotlin/by/toggi/rxbsuir/RxBsuirApplication.kt
package by.toggi.rxbsuir

import android.app.Application
import coil.ImageLoader
import coil.ImageLoaderFactory
import coil.disk.DiskCache
import dagger.hilt.android.HiltAndroidApp
import okhttp3.OkHttpClient
import javax.inject.Inject
import javax.inject.Provider

@HiltAndroidApp
class RxBsuirApplication : Application(), ImageLoaderFactory {

    @Inject
    lateinit var okHttpClient: Provider<OkHttpClient>

    override fun newImageLoader(): ImageLoader = ImageLoader.Builder(this)
        .okHttpClient(okHttpClient::get)
        .diskCache {
            DiskCache.Builder()
                .directory(cacheDir.resolve("image_cache"))
                .maxSizeBytes(50 * 1024 * 1024)
                .build()
        }
        .respectCacheHeaders(!BuildConfig.DEBUG)
        .build()
}
